src/middleware/github-server-app-middleware.ts

```diff
@@ -18,14 +18,16 @@ export const GithubServerAppMiddleware = async (req: Request, res: Response, nex
 
 		if (!gitHubServerApp) {
 			req.log.error("No GitHub app found for provided uuid.");
-			throw new Error("No GitHub app found for provided id.");
+			res.status(404).send("No GitHub app found for provided id.");
```

Comment:
According the the [doc](https://expressjs.com/en/guide/error-handling.html), we should not throw error in async code. Instead we should call `next(new Error())`.

But I tried the `next(new Error())`, it doesn't fall through to our normal error handler, not sure why but suspect it is because the error handler is in a sub router. So end result is if do `next(new Error())`, it always return 500 by the express default error handler.

So I think it okay to shortcut it to return a status code and text directly inside the middleware.

Reply:
Would it make to use `.json()` instead and return a properly-formed error obj?

`.json({error:'blahblah'})` 

?

Reply:
fixed.